### ISSUE TITLE
Option to change the default behaviour of searching + `Enter`

### ISSUE THREAD

**Issue by sapioit** [stefanXO/Tab-Manager-Plus#164]
Hi!

By default, after searching through the tabs, if we press `Enter`, all the leftover selected tabs will get pulled into a new tab, with no undo option. Could you, please, add the setting for that action to simply to to that tab? And, I would go even further, could you add to that the functionality for the up and down arrows to go through the selected tabs of the search results (like we would hover over them with the mouse), so we can press `Enter` and go to the tab we chose?

**Comment by sapioit** [comment]
When typing in the search box, being a single line, `home` and `end` can already move the cursor to before the first and after the last positions, so the `up` and `down` arrow keys could be used to move in the tabs selected, and `scroll lock` to select or deselect, as a keyboard alternative to the `right click`.

**Comment by sapioit** [comment]
Or, at least, separate the moving to tab feature from splitting to new tab feature. You could move to the first tab in the search results with `Enter`/`Return`, and split to new tabs with `Shift`+`Enter`, for example.

**Comment by sapioit** [comment]
Any update?

**Comment by sapioit** [comment]
Still hoping for this change to be implemented.

**Comment by sapioit** [comment]
4 months and 10 days later, still hoping for this to be implemented.

**Comment by mastef** [comment]
Problem with hijacking left/right keys is that they are used while typing something.
Maybe [tab] jumping to select a specific tab

PRs welcome

**Comment by sapioit** [comment]
I am perfectly alright with using `[Tab]` and `[Shift]`+`[Tab]` to navigate up and down. Also, I have not mentioned hijacking the `Left Arrow` and `Right Arrow` keys, just the `Up Arrow` and `Down Arrow` keys. `Left` and `Right` are used to navigate through the words in the search terms, once you have the extension's panel open, but `Up` and `Down` aren't because we have `Home` and `End` to get to the beginning and end (respectively) of the text in the text box, since the text box only has one line of text (I tried copy-pasting multiple lines of text, so I know). Also having one functionality for when there is a single result and a different functionality when there are multiple results creates inconsistencies in behaviour, which make the extension feel buggy, even though that behaviour is intended by the developer. 

Besides, opening the extension's panel and typing something starting with a number instead of a letter doesn't work. I know that because I sometimes have multiple tabs open for different episodes of a series to talk about them in the comments section, and I sometimes still try to search starting with a number, only to realize a second later that I need to type any letter from the name of the page then a space then the number, for the extension to search what I want to search. _Though that is less of an issue (for me) than the inconsistent behaviour of searching something and pressing enter, depending on whether there is a single search result or there are more than one results._

**Comment by sapioit** [comment]
As I said, it would likely be better to make the moving of tabs to a new window require pressing not only `Enter`, but also `Shift` or `Alt` or `Ctrl` (or `Cmd` on MacOS), including when there is a single search result, and otherwise allow us to use the `Up` and `Down` keys to navigate through the search results (or through the tabs, since we have the functionality of jumping to the current tab) followed by Enter to jump to that tab. For example, the `Ctrl` / `Cmd` key could be used to select or deselect a tab for opening in a new window, but pressing `Enter` without `Shift` would still navigate to the tab we navigated to, instead of opening a new window with said tabs.

**Comment by sapioit** [comment]
Thank you for showing interest in this feature! I hope it will be implemented soon.

**Comment by pkundrat** [comment]
> As I said, it would likely be better to make the moving of tabs to a new window require pressing not only `Enter`, but also `Shift` or `Alt` or `Ctrl` (or `Cmd` on MacOS), including when there is a single search result, and otherwise allow us to use the `Up` and `Down` keys to navigate through the search results (or through the tabs, since we have the functionality of jumping to the current tab) followed by Enter to jump to that tab. For example, the `Ctrl` / `Cmd` key could be used to select or deselect a tab for opening in a new window, but pressing `Enter` without `Shift` would still navigate to the tab we navigated to, instead of opening a new window with said tabs.

Yes - exactly. I was also proposing the similar thing on the extension page. Personally i would maybe prefer avoid using Enter altogether - as move is quite a potentially destructive operation, that would be better with its own shortcut (Ctrl-M?)

Another way how to avoid switching between several matches and keeping track which is the current one would be to use shortcuts Alt-1 to Alt-9 to switch to n-th match (would be less useful if you have more than 10 matches).

Personally i would be even better off if enter always jumped to the first match (not ideal, but still an improvement).

**Comment by pkundrat** [comment]
Would there be a way to sponsor a given feature request?

**Comment by sapioit** [comment]
I agree that that would be nice.

**Comment by duskywing** [comment]
"like we would hover over them with the mouse" What does this mean? Hovering does not do anything. Otherwise a cogent request.

**Comment by duskywing** [comment]
"Another way how to avoid switching between several matches and keeping track which is the current one would be to use shortcuts Alt-1 to Alt-9 to switch to n-th match (would be less useful if you have more than 10 matches)."

How on earth would you know which nth it is. And yes you need to go between matches but why would you want to do that, if you went past the one you're looking for you could go back. If you're regularly going between them they should be on the same window.

I agree it is a potentially destructive operation but I do not think I would hit shift accidentally. More likely I would hit ctrl+m intending to do something else in the browser.

**Comment by sapioit** [comment]
Or just show checkboxes to the left of each tab, and let us use `Ctrl`+`Up` and `Ctrl`+`Down` to move up and down, and Space to check-uncheck the checkbox, then another option as well as a button, to show a menu to choose what to do with the checked tabs.
